dbuf-typecheck/src/graph.rs

```diff
@@ -0,0 +1,154 @@
+use std::{collections::HashMap, hash::Hash};
+
+#[derive(Debug)]
+pub struct TopSortBuilder<T: Clone + Hash + Eq> {
```

Comment:
It's better to provide bounds in `impl` blocks and leave declaration-site without bounds